src/utils/__tests__/internalLinks-test.js

```diff
@@ -292,9 +292,9 @@ describe('getNarrowFromLink', () => {
 
     test('on malformed stream link: treat as old format', () => {
       const expectAsName = name => expectStream(name, [streamGeneral], name);
-      expectAsName(`${streamGeneral.stream_id}`);
-      expectAsName(`-${streamGeneral.stream_id}`);
-      expectAsName(`${streamGeneral.stream_id}nonsense-general`);
+      expectAsName(`${streamGeneral.name}`);
+      expectAsName(`-${streamGeneral.name}`);
+      expectAsName(`${streamGeneral.name}nonsense-general`);
```

Comment:
I see that changing `.stream_id` to `.name` makes the first and third of these assertions pass, when they'd otherwise fail with the new pattern.

Looking a bit closer, I think we'll want to make changes in a different direction, though. 🙂

- I think the first should just be removed: a stream ID by itself isn't malformed, it's acceptable in the new style, as I've mentioned elsewhere in this review (quoting zulip/zulip@2574efde3).
- The third... hmm, this reveals something interesting, possibly a bug! The new pattern happily takes "999nonsense-general" and thinks it's well-formed in the new style, taking 999 as the stream ID. That's wrong, at least according to the comment in the web app added in zulip/zulip@2574efde3 (quoting again):
  
  ```
    Modern stream slugs look like this, where 42
    is a stream id:
  
        42
        42-stream-name
  ```
  
  I've just posted on CZO [here](https://chat.zulip.org/#narrow/stream/6-frontend/topic/stream_data's.20slug_to_name/near/1115316), asking whether the web app has got this quite right, and what we should do. So, this test has been quite useful: it's alerted us to some behavior, after we make a change, that we might or might not want! 🙂

Reply:
So, I'll wait for their response before making any changes to this file then?

Reply:
I have removed both the 1st and the 3rd assertions. Is that okay?